File: Matlab/Net.m
function lgraph = Net()
lgraph = layerGraph();
tempLayers = [
%     imageInputLayer([28 28 1],"Name","imageinput")
%     convolution2dLayer([1 10],5,"Name","conv_1","Padding","same")
%     batchNormalizationLayer
%     maxPooling2dLayer([2 2],"Name","maxpool_1","Padding","same")
%     reluLayer("Name","relu_1")
%     convolution2dLayer([10 20],5,"Name","conv_2","Padding","same")
%     batchNormalizationLayer
%     dropoutLayer(0.5,"Name","dropout")
%     
%     maxPooling2dLayer([2 2],"Name","maxpool_2","Padding","same")
%     reluLayer("Name","relu_2")
%     fullyConnectedLayer(320,"Name","fc_1")
%     batchNormalizationLayer
%     fullyConnectedLayer(50,"Name","fc_2")
%     reluLayer("Name","relu_3")
%     dropoutLayer(0.5,"Name","dropout_2")
%     fullyConnectedLayer(10,"Name","fc")
%     softmaxLayer("Name","softmax")
%     classificationLayer("Name","classoutput")


%     imageInputLayer([28 28 1])
%     
%     convolution2dLayer(3,8,'Padding','same')
%     batchNormalizationLayer
%     reluLayer
%     
%     maxPooling2dLayer(2,'Stride',2)
%     
%     convolution2dLayer(3,16,'Padding','same')
%     batchNormalizationLayer
%     reluLayer
%     
%     maxPooling2dLayer(2,'Stride',2)
%     
%     convolution2dLayer(3,32,'Padding','same')
%     batchNormalizationLayer
%     reluLayer
%     
%     fullyConnectedLayer(10)
%     softmaxLayer
%     classificationLayer

    imageInputLayer([28 28 1])
    
    convolution2dLayer(5,10,"Name","conv_1","Padding","same")
    batchNormalizationLayer
    maxPooling2dLayer([2 2],"Name","maxpool_1","Padding","same")
    reluLayer("Name","relu_1")
    convolution2dLayer(5,20,"Name","conv_2","Padding","same")
    batchNormalizationLayer
    dropoutLayer(0.5,"Name","dropout_1")
    maxPooling2dLayer([2 2],"Name","maxpool_2","Padding","same")
    reluLayer("Name","relu_2")

    fullyConnectedLayer(50,"Name","fc_1")
    reluLayer("Name","relu_3")
    dropoutLayer(0.5,"Name","dropout_2")
    fullyConnectedLayer(10,"Name","fc_2")
    softmaxLayer("Name","softmax")
    classificationLayer("Name","classoutput")
    ];
lgraph = addLayers(lgraph,tempLayers);

% 清理辅助变量
clear tempLayers;

end

File: Python/net.py
import torch.nn as nn
import torch.nn.functional as F
 
class Net(nn.Module):
    def __init__(self):
        super(Net, self).__init__()
        '''定义相关的计算层'''
        # 定义一个卷积核为1×10的卷积层
        self.conv1 = nn.Conv2d(1, 10, kernel_size=5)
        # 定义一个卷积核为10×20的卷积层
        self.conv2 = nn.Conv2d(10, 20, kernel_size=5)
        # 定义一个二维的Dropout2层，防止模型训练时过拟合
        self.conv2_drop = nn.Dropout2d()
        # 定义全连接层
        self.fc1 = nn.Linear(320, 50)
        self.fc2 = nn.Linear(50, 10)
    def forward(self, x):
        # 一次卷积运算
        x = self.conv1(x)
        x = F.max_pool2d(x, 2)
        x = F.relu(x)
        # 二次卷积运算
        x = self.conv2(x)
        x = self.conv2_drop(x, 2)
        x = F.max_pool2d(x)
        x = F.relu(x)
        # 设置数据长度
        x = x.view(-1, 320)
        # 一次全连接
        x = self.fc1(x)
        x = F.relu(x)
        x = F.dropout(x, training=self.training)
        # 二次全连接
        x = self.fc2(x)
        # 分类模型调用分类结果处理函数
        return F.log_softmax(x) 